src/ADAL.PCL/AdalHttpClient.cs

```diff
@@ -132,10 +155,11 @@ public AdalHttpClient(string uri, CallState callState)
             return typedResponse;
         }
 
+
         private bool isDeviceAuthChallenge(string endpointType, IHttpWebResponse response, bool respondToDeviceAuthChallenge)
         {
             return PlatformPlugin.DeviceAuthHelper.CanHandleDeviceAuthChallenge &&
-                   respondToDeviceAuthChallenge &&
+                   respondToDeviceAuthChallenge && response!=null &&
```

Comment:
The check is added for null pointer exception.Hence need to add the check before response is dereferenced.